src/fragments/lib-v1/ssr/nextjs/getting-started.mdx

```diff
@@ -0,0 +1,338 @@
+
+```typescript
+import { cookies } from 'next/headers';
+import { runWithAmplifyServerContext } from '@aws-amplify/adapter-nextjs';
+import { getCurrentUser } from '@aws-amplify/auth/server';
+
+// This page always dynamically renders per request
+export const dynamic = 'force-dynamic';
+
+export default async function AuthGetCurrentUserServer() {
+  try {
+    const currentUser = await runWithAmplifyServerContext({
+      nextServerContext: { cookies },
+      operation: (contextSpec) => getCurrentUser(contextSpec),
+    });
+
+    return (
+      <AuthFetchResult
+        description="The API is called on the server side."
+        data={currentUser}
+      />
+    );
+  } catch (error) {
+    console.error(error);
+    return <p>Something went wrong...</p>;
+  }
+}
+```
+
+##### Static Rendering
+
+Static rendering doesn’t require a user session, so you can specify the
+`nextServerContext` parameter as `null`. This is useful for some use cases,
+for example, when you are using the Storage API with guest access (if you have
+enabled it).
+
+```typescript
+import { runWithAmplifyServerContext } from '@aws-amplify/adapter-nextjs';
+import { getUrl } from 'aws-amplify/storage/server';
+import Image from 'next/image';
+
+// Re-render this page every 60 minutes
+export const revalidate = 60 * 60; // in seconds
+
+export default async function StaticallyRenderedPage() {
+  try {
+    const splashUrl = await runWithAmplifyServerContext({
+        nextServerContext: null,
+        operation: (contextSpec) =>
+        getUrl(contextSpec, {
+            key: 'splash.png',
+        }),
+    });
+    
+    return (
+      <Image
+        src={splashUrl.url.toString()}
+        alt="Splash Image"
+        width={500}
+        height={500}
+      />
+    );
+  } catch (error) {
+    console.error(error);
+    return <p>Something went wrong...</p>;
+  }
+}
+```
+
+<Callout>
+
+**NOTE:** the URL returned by the `getUrl` API expires in the above example.
+You may want to specify the `revalidate` to rerender the page to ensure the
+URL gets regenerated.
+
+</Callout>
+
+#### In Route Handlers
+
+Take implementing an API route that enables `GET /apis/get-current-user`.
+
+```typescript
+import { runWithAmplifyServerContext } from '@aws-amplify/adapter-nextjs';
+import { getCurrentUser } from 'aws-amplify/auth/server';
+import { cookies } from 'next/headers';
+import { NextResponse } from 'next/server';
+
+export async function GET() {
+  const user = await runWithAmplifyServerContext({
+    nextServerContext: { cookies },
+    operation: (contextSpec) => getCurrentUser(contextSpec),
+  });
+
+  return NextResponse.json({ user });
+}
+```
+
+When you call `fetch('/apis/get-current-user')` it return a payload that contain
+the `user` data.
+
+### With Next.js Pages Route
+
+#### In `getServerSideProps`
+
+Take getting current user data extracted from the request, for passing the server
+side props into the `Page` component.
```

Comment:
```suggestion
The following example extracts current user data from the request and provides them to a server component via its props.
```

Reply:
Updated as "The following example extracts current user data from the request and provides
them to a page react component via its props."

server component is not a term with in the Pages Router.